### DISCUSSION TITLE
DOCX reader - How to create table caption in non-English Microsoft Word?

### DISCUSSION THREAD

**Discussion by rgaiacs** [jgm/pandoc#9515] [Q&A]
I have my Microsoft Word in German and my document in English.

![Screenshot 2024-02-26 170602](https://github.com/jgm/pandoc/assets/1506457/4994303b-e428-4f6f-9ac0-da09237044b9)

I create a table using the Microsoft Word built-in interface.

![Screenshot 2024-02-26 170752](https://github.com/jgm/pandoc/assets/1506457/60d7567c-f811-446a-82a4-b075adf0935a)

And I add a caption using the Microsoft Word built-in dialogue window.

![Screenshot 2024-02-26 170852](https://github.com/jgm/pandoc/assets/1506457/4a8d8f97-b98d-4fdf-a66b-c12f2081e247)

Because my document is in English, Word automatically set the caption to "Table".

The final minimal working example is [mwe-using-german-word.docx](https://github.com/jgm/pandoc/files/14407245/mwe-using-german-word.docx).

When I run `pandoc --from docx --to html mwe-using-german-word.docx`, the output is

```
<p>Lorem ipsum</p>
<p>Table 1 Example</p>
<table>
<colgroup>
<col style="width: 50%" />
<col style="width: 50%" />
</colgroup>
<thead>
<tr class="header">
<th>A</th>
<th>B</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>C</td>
<td>D</td>
</tr>
</tbody>
</table>
```

instead of

```
<p>Lorem ipsum</p>
<table>
<caption><p>Example</p></caption>
<colgroup>
<col style="width: 50%" />
<col style="width: 50%" />
</colgroup>
<thead>
<tr class="header">
<th>A</th>
<th>B</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>2</td>
</tr>
</tbody>
</table>
```

that is produced by the same command (`pandoc --from docx --to html`) but using [mwe-using-english-word.docx](https://github.com/jgm/pandoc/files/14407372/mwe-using-english-word.docx) as input.

# XML of non-English Document

The caption is

```
    <w:p w14:paraId="1FADD07B" w14:textId="3660CC9A" w:rsidR="00917377" w:rsidRDefault="00917377" w:rsidP="00917377">
      <w:pPr>
        <w:pStyle w:val="Beschriftung"/>
        <w:keepNext/>
      </w:pPr>
      <w:r>
        <w:t xml:space="preserve">Table </w:t>
      </w:r>
      <w:r>
        <w:fldChar w:fldCharType="begin"/>
      </w:r>
      <w:r>
        <w:instrText xml:space="preserve"> SEQ Table \* ARABIC </w:instrText>
      </w:r>
      <w:r>
        <w:fldChar w:fldCharType="separate"/>
      </w:r>
      <w:r>
        <w:rPr>
          <w:noProof/>
        </w:rPr>
        <w:t>1</w:t>
      </w:r>
      <w:r>
        <w:fldChar w:fldCharType="end"/>
      </w:r>
      <w:r>
        <w:t xml:space="preserve"> </w:t>
      </w:r>
      <w:proofErr w:type="spellStart"/>
      <w:r>
        <w:t>Example</w:t>
      </w:r>
      <w:proofErr w:type="spellEnd"/>
    </w:p>
```

# XML of English Document

```
    <w:p w14:paraId="5DE3A68F" w14:textId="153D5F3C" w:rsidR="000E6255" w:rsidRDefault="000E6255" w:rsidP="000E6255">
      <w:pPr>
        <w:pStyle w:val="Caption"/>
        <w:keepNext/>
      </w:pPr>
      <w:r>
        <w:t xml:space="preserve">Table </w:t>
      </w:r>
      <w:fldSimple w:instr=" SEQ Table \* ARABIC ">
        <w:r>
          <w:rPr>
            <w:noProof/>
          </w:rPr>
          <w:t>1</w:t>
        </w:r>
      </w:fldSimple>
      <w:r>
        <w:t xml:space="preserve"> Example</w:t>
      </w:r>
    </w:p>
```

**Comment by jgm** [comment]
So I guess the issue is that the style Word uses has different names, "Caption" or "Beschriftung", depending on the localization of your Word version.  And we're looking for "Caption."

**Comment by jgm** [comment]
In styles.xml we have:
```xml
<w:style w:type="paragraph" w:styleId="Beschriftung">
<w:name w:val="caption"/>
```
So what we need to do, instead of checking the styleId, is *look up* the style id and check the style's `<w:name>` element to see if it is "caption".

**Comment by jgm** [comment]
Please submit an issue with this suggestion, linking to this discussion.
